builtin.c

```diff
@@ -63,6 +63,30 @@ static void S_finish_export_lexical(pTHX)
     LEAVE;
 }
 
+XS(XS_builtin_load);
+XS(XS_builtin_load)
+{
+    dXSARGS;
+
+    if (items != 1) {
+        croak_xs_usage(cv, "arg");
+    }
+
+    SV *module = TOPs;
+
+    if (!SvOK(module)) {
```

Comment:
SvOK() isn't valid before checking for magic:
```
tony@venus:.../git/perl5$ cat ../20316.pl
use v5.36.0;
use builtin 'load';
my $x;
tie my $y, TestTie => \$x;
$x = "Storable";
eval { load $y; 1 } or print $@;
print ">>$y<<\n";

package TestTie;
sub TIESCALAR {
    bless $_[1], $_[0];
}
sub FETCH {
    ${$_[0]}
}

tony@venus:.../git/perl5$ ./perl -Ilib ../20316.pl
Built-in function 'builtin::load' is experimental at ../20316.pl line 6.
Undefined argument to builtin::load at ../20316.pl line 6.
>>Storable<<
tony@venus:.../git/perl5$
```

You can `SvGETMAGIC(module)` just before the SvOK() and then use newSVpv_nomg() to duplicate it while avoiding an extra magic call.